test/basics.jl

```diff
@@ -0,0 +1,45 @@
+string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UniStr]
+
+@testset "constructors" begin
+    for strtype in string_types
+        @test convert(strtype, [0x61,0x62,0x63,0x21]) == "abc!"
+        @test convert(strtype, "abc!") == "abc!"
+
+        @test isempty(string())
+        @test eltype(GenericString) == Char
+        @test start("abc") == 1
+        @test cmp("ab","abc") == -1
+        @test "abc" === "abc"
+        @test "ab"  !== "abc"
+        @test string("ab", 'c') === "abc"
+        @test string() === ""
```

Comment:
This doesn't test anything for the Str types, move out of the loop.
I'll try to think of something that is similar

Reply:
What about `@test start("abc") == 1`?
Look like, I'd need to replace `"abc"` with `str_abc`, right?